File: src/_06_extension_functions/ExtensionFunctions.java
package _06_extension_functions;

import java.util.ArrayList;
import java.util.Arrays;

class ExtensionFunctions {
    public static void main(String[] args) {
        ArrayList<Integer> list = new ArrayList<>(Arrays.asList(1, 2, 3));
        Collections.swap(list,0, 2);
        System.out.println(list);
    }
}

class Collections {
    public static <T> void swap(ArrayList<T> list, int index1, int index2) {
        var tmp = list.get(index1);
        list.set(index1, list.get(index2));
        list.set(index2, tmp);
    }
}

File: src/_06_extension_functions/ExtensionFunctions.kt
package _06_extension_functions


fun <T> MutableList<T>.swap(index1: Int, index2: Int) {
    val tmp = this[index1] // 'this' corresponds to the list
    this[index1] = this[index2]
    this[index2] = tmp
}


fun main() {
    val list = mutableListOf(1, 2, 3)
    list.swap(0, 2)
    println(list)
}
